internal/controlplane/handlers_profile.go

```diff
@@ -644,3 +554,273 @@ func (s *Server) UpdateProfile(ctx context.Context,
 		Profile: updatedProfile,
 	}, nil
 }
+
+// GetProfileStatusByName retrieves profile status by name
+func (s *Server) GetProfileStatusByName(
+	ctx context.Context,
+	in *minderv1.GetProfileStatusByNameRequest,
+) (*minderv1.GetProfileStatusByNameResponse, error) {
+	ctx = context.WithValue(ctx, requestKey, in)
+	resp, err := s.getProfileStatus(
+		ctx,
+		engcontext.EntityFromContext(ctx).Project.ID,
+		in.Name,
+		s.store.GetProfileStatusByNameAndProject,
+		nil, // No ID function required
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return &minderv1.GetProfileStatusByNameResponse{
+		ProfileStatus:        resp.ProfileStatus,
+		RuleEvaluationStatus: resp.RuleEvaluationStatus,
+	}, nil
+}
+
+// GetProfileStatusById retrieves profile status by ID
+func (s *Server) GetProfileStatusById(
+	ctx context.Context,
+	in *minderv1.GetProfileStatusByIdRequest,
+) (*minderv1.GetProfileStatusByIdResponse, error) {
+	ctx = context.WithValue(ctx, requestKey, in)
+
+	resp, err := s.getProfileStatus(
+		ctx,
+		engcontext.EntityFromContext(ctx).Project.ID,
+		uuid.MustParse(in.Id),
+		nil, // No name function required
+		s.store.GetProfileStatusByIdAndProject,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// Cast response to GetProfileStatusByIdResponse
+	return &minderv1.GetProfileStatusByIdResponse{
+		ProfileStatus:        resp.ProfileStatus,
+		RuleEvaluationStatus: resp.RuleEvaluationStatus,
+	}, nil
+}
+
+func (s *Server) getProfileStatus(ctx context.Context,
+	projectID uuid.UUID,
+	profileIdentifier interface{},
+	getProfileByNameFunc func(context.Context,
+		db.GetProfileStatusByNameAndProjectParams,
+	) (db.GetProfileStatusByNameAndProjectRow, error),
+	getProfileByIdFunc func(context.Context,
+		db.GetProfileStatusByIdAndProjectParams,
+	) (db.GetProfileStatusByIdAndProjectRow, error)) (*minderv1.GetProfileStatusResponse, error) {
+
+	entityCtx := engcontext.EntityFromContext(ctx)
+
+	// Validate project
+	if err := entityCtx.ValidateProject(ctx, s.store); err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "error in entity context: %v", err)
+	}
+
+	// Retrieve profile status
+	var dbProfileStatus interface{}
+	var getProfileErr error
+
+	switch id := profileIdentifier.(type) {
+	case string:
```

Comment:
I worry that the switch statement here is fragile and makes it harder to follow the codepath.
Do you see a way to call `GetProfileStatusByNameAndProjectParams` or `GetProfileStatusByIdAndProjectParams` respectively in their corresponding top level functions and then pass the results to the helper?

Reply:
Have refactored the code , to make the flow more simple to follow